src/app_service/service/accounts/service.nim

```diff
@@ -615,17 +643,21 @@ QtObject:
           "DiscV5BootstrapNodes": @[]
         }
 
-      let response = status_account.login(account.name, account.keyUid, account.kdfIterations, hashedPassword, thumbnailImage,
-        largeImage, $nodeCfg)
-      var error = "response doesn't contain \"error\""
-      if(response.result.contains("error")):
-        error = response.result["error"].getStr
-        if error == "":
-          debug "Account logged in"
-          self.loggedInAccount = account
-          self.setLocalAccountSettingsFile()
 
-      return error
+      let isOldHashPassword = self.verifyDatabasePassword(account.keyUid, hashPassword(password, lower=false))
+      if isOldHashPassword:
+        discard status_privacy.lowerDatabasePassword(account.keyUid, password)
```

Comment:
If there's a failure calling this function, will an exception be raised and captured in line 662? or will it fail silently? The reason why I ask is because in `src/backend/privacy.nim` I'm not sure how a RPCException could be generated in case of errors

Reply:
yes, for this function it either succeed or raise an error, if I understand correctly.

Also before calling this I validate that the password is correct by calling: verifyDatabasePassword which guarantee me a success